ananse/enhancer_binding.py

```diff
@@ -0,0 +1,497 @@
+                rest = line[3:] if not output_bed3 else []
+
+                chrm_len = chrom_sizes[chrm]
+                if width == end - start:
+                    nstart = str(start)
+                    nend = str(end)
+                elif chrm_len <= width:
+                    if not fix_outliers:
+                        continue
+                    nstart = str(0)
+                    nend = str(chrm_len)
+                else:
+                    summit = get_summit(start, end, line[summit_col])
+                    if not fix_outliers:
+                        nstart = str(summit - half_seqlen)
+                        nend = str(summit + half_seqlen)
+                        if int(nstart) < 0 or int(nend) > chrm_len:
+                            continue
+                    else:
+                        # adjust the summit for the chromosome boundaries
+                        summit = max(summit, 0 + half_seqlen)
+                        summit = min(summit, chrm_len - half_seqlen)
+
+                        nstart = str(summit - half_seqlen)
+                        nend = str(summit + half_seqlen)
+
+                new.write("\t".join([chrm, nstart, nend] + rest) + "\n")
+
+        if missing_chrm and verbose:
+            print(
+                "The following contigs were present in "
+                + f"'{os.path.basename(bed_in)}', "
+                + "but were missing in the genome file: "
+                + f"{', '.join(list(set(missing_chrm)))}\n"
+            )
+        return bed_out
+
+    def run(self, outfile, width=200, force=False):
+        if force or not os.path.exists(outfile):
+            if self.verbose:
+                print("Combining bed files")
+            tmpdir = tempfile.mkdtemp()
+            try:
+                list_of_beds = []
+                for peakfile in self.list_of_peakfiles:
+                    # use narrowPeak Peak location for region centering if possible
+                    is_np = self.is_narrowpeak(peakfile)
+                    resized_peakfile = os.path.join(tmpdir, os.path.basename(peakfile))
+
+                    # resize each BED region to 200 BP
+                    self.bed_resize(
+                        genome=self.genome,
+                        bed_in=peakfile,
+                        bed_out=resized_peakfile,
+                        width=width,
+                        narrowpeak=is_np,
+                        verbose=self.verbose,
+                    )
+                    bed_sort(resized_peakfile)
+                    list_of_beds.append(resized_peakfile)
+
+                # merge resized beds into one
+                merged_bed = os.path.join(tmpdir, "merged")
+                bed_merge(list_of_beds=list_of_beds, merged_bed=merged_bed)
+
+                bed_sort(merged_bed)
+                shutil.copy2(merged_bed, outfile)
+            finally:
+                shutil.rmtree(tmpdir, ignore_errors=True)
+
+
+class CombineBamFiles:
```

Comment:
I am just skimming through the code. But are we adding the bam files together? Isn't it better to take the mean/median/... from samples? After normalizing for sequencing depth?

Reply:
Wasn't the idea to use mosdepth?

Reply:
mean per-region given a BED file of regions would work as well! I'll have a look at re-reimplementing this

Reply:
I would be in favor of at least trying mosdepth. Maybe the best course of action would be to add that as an option to `coverage_table` in gimme? Then all tools that use that can profit :).

Reply:
`mosdepth` is implemented. We can come back to this if `coverage_table` turns more useful.